src/components/Personalization/dom-actions/addNonceToInlineStyleElements.js

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { selectNodes } from "../../../utils/dom";
+import { SRC, STYLE } from "../../../constants/tagName";
+import { getAttribute, getNonce } from "./dom";
+
+const is = (element, tagName) => element.tagName === tagName;
+const isInlineStyleElement = element =>
+  is(element, STYLE) && !getAttribute(element, SRC);
+
+export default fragment => {
+  const styleNodes = selectNodes(STYLE, fragment);
+  const { length } = styleNodes;
+  const nonce = getNonce();
+  if (!nonce) {
+    return;
+  }
+  /* eslint-disable no-continue */
+  for (let i = 0; i < length; i += 1) {
+    const element = styleNodes[i];
+    if (!isInlineStyleElement(element)) {
+      continue;
+    }
+    element.nonce = nonce;
```

Comment:
When https://developer.mozilla.org/en-US/docs/Web/API/HTMLOrForeignElement/nonce says:

> In the past, not all browsers supported the nonce IDL attribute, so a workaround is to try to use getAttribute as a fallback...

I don't know if that's only for _reading_ the nonce value or also setting it. If old browsers don't allow you to _set_ a nonce IDL attribute, then `element.nonce = nonce;` probably won't have the effect we're after. Anyway, that might take some digging to figure out. I'm okay if you decide to leave it as-is. I'd rather leave it as-is than add a fallback when we don't know if we actually need a fallback.